tests/test_dispatch_client.py

```diff
@@ -2,4 +2,97 @@
 # Copyright © 2024 Frequenz Energy-as-a-Service GmbH
 
 """Tests for the frequenz.client.dispatch package."""
+
+from contextlib import asynccontextmanager
+from typing import AsyncGenerator
+
 import pytest
+
+# pylint: disable=no-name-in-module
+from frequenz.api.dispatch.v1 import dispatch_pb2_grpc
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    Dispatch,
+    DispatchCreateRequest,
+    DispatchDeleteRequest,
+    DispatchGetRequest,
+    DispatchList,
+    DispatchListRequest,
+    DispatchUpdateRequest,
+)
+from google.protobuf.empty_pb2 import Empty
+from grpc import aio
+
+from frequenz.client.dispatch import Client
+
+# pylint: enable=no-name-in-module
+
+
+class DispatchServicer(dispatch_pb2_grpc.MicrogridDispatchServiceServicer):
+    """Dispatch servicer for testing."""
+
+    def ListMicrogridDispatches(
+        self, request: DispatchListRequest, context: aio.ServicerContext  # type: ignore
+    ) -> DispatchList:
+        """List microgrid dispatches."""
+        return DispatchList()
+
+    def CreateMicrogridDispatch(
+        self,
+        request: DispatchCreateRequest,
+        context: aio.ServicerContext,  # type: ignore
+    ) -> Empty:
+        """Create a new dispatch."""
+        return Empty()
+
+    def UpdateMicrogridDispatch(
+        self,
+        request: DispatchUpdateRequest,
+        context: aio.ServicerContext,  # type: ignore
+    ) -> Empty:
+        """Update a dispatch."""
+        return Empty()
+
+    def GetMicrogridDispatch(
+        self,
+        request: DispatchGetRequest,
+        context: aio.ServicerContext,  # type: ignore
+    ) -> Dispatch:
+        """Get a single dispatch."""
+        return Dispatch()
+
+    def DeleteMicrogridDispatch(
+        self,
+        request: DispatchDeleteRequest,
+        context: aio.ServicerContext,  # type: ignore
+    ) -> Empty:
+        """Delete a given dispatch."""
+        return Empty()
+
+
+ServicerType = AsyncGenerator[tuple[aio.Server, int], None]
+
+
+@asynccontextmanager
+async def grpc_servicer() -> AsyncGenerator[tuple[aio.Server, str], None]:
+    """Async context manager to setup and teardown a gRPC server for testing."""
+    server = aio.server()
+    dispatch_servicer = DispatchServicer()
+    dispatch_pb2_grpc.add_MicrogridDispatchServiceServicer_to_server(
+        dispatch_servicer, server
+    )
+    port = server.add_insecure_port("localhost:0")
+    await server.start()
+    try:
+        yield server, f"localhost:{port}"
+    finally:
+        await server.stop(None)
+
+
+@pytest.mark.asyncio
```

Comment:
This mark shouldn't be necessary, usually pytest detects it automatically. Any reason to include it?

Reply:
It was ChatGPTed :angel: